### ISSUE TITLE
UI Added SSH key showing OFF after logout (need to manually turn ON everytime)

### ISSUE THREAD

**Issue by geoffroyjabouley** [cockpit-project/cockpit#8613]
Hello

i don't know where to post that, so trying here. Maybe it is not an issue but a feature...

We have 2 machines, both using same `user` account:
* Cockpit_main_host
* Target_to_monitor_host

**Configuration**
1. Create a dedicated SSH key pair, without passphrase, for my cockpit monitoring setup: `myproject_rsa / myproject_rsa.pub`
1. Copy `myproject_rsa.pub` into `Target_to_monitor_host:/home/user/.ssh/authorized_keys` --> Success to remotely ssh from `Cockpit_main_host` to `Target_to_monitor_host`
1. Now, after cockpit v157 installation on CentOS, login in `Cockpit_main_host:9090/dashboard`
1. Manually add the SSH private key `myproject_rsa` using `Cockpit UI Account / Authentication / Add key`
1. Create a new server `Target_to_monitor_host`, setting authentication to use previously configured authentication.

**Result**
Connection to `Target_to_monitor_host `is established
Anytime i logout / login from the Cockpit UI, my `Target_to_monitor_host` appears not reachable, and i have to manually go into `Account / Authentication` to turn `myproject_rsa` SSH key ON (showing OFF)

**Current workaround**
it seems that ssh-agent used by Cockpit is by default loading just a bunch of local SSH keys
* ~/.ssh/id_rsa
* ~/.ssh/id_dsa
* ~/.ssh/id_ed25519
* ~/.ssh/id_ecdsa

So creating a _soft link_ from `myproject_rsa` to `id_rsa` solved this problem.

**Question**
Why Cockpit is not starting a ssh-agent with UI added SSH keys automatically added into it, using ssh-add?
Is it normal behavior, for security reasons? Are we misconfiguring SSH credentials for Cockpit?

**Comment by a-tyler** [comment]
Sorry for commenting after the issue is closed and please forgive my ignorance but...
**geoffroyjabouley said:**
> 1. Create a dedicated SSH key pair, without passphrase, for my cockpit monitoring setup: myproject_rsa / myproject_rsa.pub

If an ssh key is generated without a passphrase would that not indicate that the key is already unencrypted? I mean the command line ssh client does not prompt for a passphrase for keys that do not require one correct?

**Comment by martinpitt** [comment]
@a-tyler : That's correct. With such an unprotected key you really need to know what you are doing.

**Comment by ZEROF** [comment]
Hi,
I was not sure if I need to open new case, but this is kind annoying. When I add more then one node to dashboard, I get error after every logout/login when I want to access to other nodes:

    Couldn't connect to the machine
    Login failed

I need to go to user setting, add key again, and then I can login to other server and get monitoring etc.
Using user/password for me it's evenless secure, because enabling password for user is what we don't want. 

Does anybody have solution for this?

**Comment by stfloyd** [comment]
I am running into this issue as well, would be nice if anyone had a workaround or solution.

**Comment by mvollmer** [comment]
What about specifying the key in ~/.ssh/config for the host that you need it for.  If it is unencrypted, it should just be used without having to add it to the agent, I'd think.

```
Host 127.0.0.2
     User root
     IdentityFile ~/work/cockpit/bots/machine/identity
```
